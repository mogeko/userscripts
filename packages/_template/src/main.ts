console.log("Hello World");

export type {};
