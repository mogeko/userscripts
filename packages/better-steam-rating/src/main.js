function hideEle(item) {
  let parentNode = item.parentNode;
  let maxLoop = 15;

  while (parentNode.tagName !== "A" && maxLoop > 0) {
    parentNode = parentNode.parentNode;
    maxLoop--;
  }

  item.dataset.isCheck = "1";
  parentNode.style.display = "none";
}

function checkFn() {
  const minRealNum = 100;
  const items = document.querySelectorAll(".responsive_search_name_combined");

  items.forEach((item2) => {
    if (item2.dataset.isCheck) return;

    let days = 0;
    const item = item2.querySelector(".search_review_summary");
    const dateItem = item2.querySelector(".search_released");

    if (dateItem) {
      const dateText = dateItem.innerText;
      const dateArr = dateText
        .replace("年", "/")
        .replace("月", "/")
        .replace("日", "")
        .split("/");
      if (dateArr.length > 2) {
        const gameDate = new Date();
        gameDate.setFullYear(dateArr[0]);
        gameDate.setMonth(dateArr[1] - 1);
        gameDate.setDate(dateArr[2]);

        days = (new Date().getTime() - gameDate.getTime()) / 86400000;
        days = parseInt(days);
      }
    }

    if (!item) {
      hideEle(item2);
      return;
    }

    const content = item.dataset.tooltipHtml;
    const contentArr = content.split("<br>");

    if (contentArr.length !== 2) {
      console.log(content);
      return;
    }

    const last = contentArr[1];
    const lastArr = last.split(" ");

    if (lastArr < 2) {
      console.log(content);
      return;
    }

    const num = lastArr[0].replace(/\,/g, "");
    const rate = lastArr[2].replace("%", "");
    const realNum = parseInt((num * rate) / 100);

    if (isNaN(realNum)) {
      console.log(content);
      return;
    }

    let realRate = "-";

    if (realNum && days) {
      realRate = realNum / days;
      realRate = realRate > 10 ? realRate.toFixed(0) : realRate.toFixed(1);
    }

    const innerHtml = `<span style="padding-left: 0.5em;width: 2em;display: inline-block;">${rate}</span><span style="width: 4.5em;display:inline-block;">${parseInt(
      (num * rate) / 100,
    )}</span><span style="width: 3em;display:inline-block;">${days}</span><span style="width:3em;display:inline-block;">${realRate}</span>`;

    item.innerHTML = innerHtml;
    item.style.width = "13em";
    item.style.marginLeft = "-13em";
    item.style.backgroundColor = "#000";
    item.style.backgroundImage = "none";
    item.style.textAlign = "left";
    item.style.color = "#fff";
    item2.dataset.isCheck = "1";

    if (realNum && realNum < minRealNum && realRate !== "-" && realRate < 0.1) {
      hideEle(item);
    }
  });
}

setInterval(() => {
  checkFn();
}, 2000);

export {};
