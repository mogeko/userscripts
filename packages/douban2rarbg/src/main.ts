const META_DATA = {
  资源: {
    RARBG: "https://rarbg.to/torrents.php?imdb=%i&order=seeders&by=DESC",
    "RARBG (Mirror)":
      "https://rarbgmirror.com/torrents.php?imdb=%i&order=seeders&by=DESC",
    TorrentGalaxy: "https://torrentgalaxy.to/torrents.php?search=%i",
    TPB: "https://thepiratebay.org/search.php?q=%i",
  },
  字幕: {
    opensubtitles:
      "https://www.opensubtitles.org/zh/search/imdbid-%x/sublanguageid-all/moviename-%i",
    SubHD: "https://subhd.tv/search/%d",
    字幕库: "https://so.zimuku.org/search?q=%i",
    R3SUB: "https://r3sub.com/search.php?s=%i",
    点点字幕: "http://www.ddzimu.com/download/xslist.php?key=%d",
  },
};

(function () {
  const metaRoot = document.querySelector("#info");
  const imdb = metaRoot?.textContent?.match(/tt[0-9]{4,}/)?.[0];
  const doubanID = document.location.toString().split("/")[4];

  if (!imdb || !doubanID) return;

  Object.entries(META_DATA).forEach(([key, sites]) => {
    const metaNode = document.createElement("span");
    const plNode = document.createElement("span");
    const attrsNode = document.createElement("span");
    const br = document.createElement("br");

    plNode.setAttribute("class", "pl");
    plNode.textContent = `${key}: `;

    const links = Object.entries(sites).map(([title, template]) => {
      const handleTemplate = (template: string) => {
        const [i, d, x] = [imdb, doubanID, imdb.replace(/^tt/, "")];

        return template.replace("%i", i).replace("%d", d).replace("%x", x);
      };
      const link = document.createElement("a");

      link.setAttribute("href", handleTemplate(template));
      link.setAttribute("target", "_blank");
      link.textContent = title;

      return link;
    });

    attrsNode.setAttribute("class", "attrs");
    links.forEach((link, index, array) => {
      attrsNode.appendChild(link);
      if (index !== array.length - 1) {
        attrsNode.innerHTML += " / ";
      }
    });

    metaNode.appendChild(plNode);
    metaNode.appendChild(attrsNode);

    metaRoot.appendChild(metaNode);
    metaRoot.appendChild(br);
  });
})();

export {};
