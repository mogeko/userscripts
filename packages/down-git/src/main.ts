/**
 * <div role="row">
 *  <div><svg class={{icon}}/></div>,
 *  <div><span><a>{{name}}</a></span></div>,
 *  <div><span><a>{{message}}</a></span></div>,
 *  <div><span>{time}</span></div>,
 * </div>
 */

const SVG_ICON = `<svg class="octicon octicon-download hx_color-icon-directory" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M7.47 10.78a.75.75 0 001.06 0l3.75-3.75a.75.75 0 00-1.06-1.06L8.75 8.44V1.75a.75.75 0 00-1.5 0v6.69L4.78 5.97a.75.75 0 00-1.06 1.06l3.75 3.75zM3.75 13a.75.75 0 000 1.5h8.5a.75.75 0 000-1.5h-8.5z"/></svg>`;

const DOWN_GIT = "https://minhaskamal.github.io/DownGit";
const JSDELIVR = "https://cdn.jsdelivr.net";
// /caiyongji/emoji-list/blob/master/LICENSE
function downloader(localUrl: string) {
  const meta = localUrl?.split("/").filter((str) => str);
  const [_a, _b, user, repo, flag, branch, ...rest] = meta;

  if (flag === "tree") {
    return `${DOWN_GIT}/#/home?url=${localUrl}`;
  }
  if (flag === "blob") {
    return `${JSDELIVR}/gh/${user}/${repo}@${branch}/${rest.join("/")}`;
  }
  return localUrl;
}

function setButton(urlNode: HTMLAnchorElement) {
  const wrapNode = document.createElement("div");
  const linkNode = document.createElement("a");
  const url = downloader(urlNode.href);

  wrapNode.setAttribute("class", "mr-3 flex-shrink-0");
  linkNode.setAttribute("href", url ?? "");
  linkNode.setAttribute("title", `Download ${urlNode.innerHTML}`);
  linkNode.innerHTML = SVG_ICON;
  wrapNode.appendChild(linkNode);

  return wrapNode;
}

requestIdleCallback(() => {
  for (const node of document.querySelectorAll("div.Box-row")) {
    const urlNode = node.querySelector("a");
    const anchorNode = node.querySelector("div.text-right");

    if (!urlNode || urlNode.querySelector("span")) return;

    node.insertBefore(setButton(urlNode), anchorNode);
  }
});

export type {};
