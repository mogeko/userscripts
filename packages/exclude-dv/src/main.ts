const HDR = "https://dyncdn.me/static/20/images/categories/cat_new52.gif";

document.querySelectorAll("table.lista2t tr.lista2").forEach((tr) => {
  const link = tr.querySelector<HTMLLinkElement>("td:nth-child(2) > a");
  const cover = tr.querySelector<HTMLImageElement>("td:nth-child(1) img");

  if (link?.title?.includes(".DV.") && cover?.src === HDR) {
    console.log("[Exclude DV] Remove: ", link?.title);
    tr.remove();
  }
});

export {};
