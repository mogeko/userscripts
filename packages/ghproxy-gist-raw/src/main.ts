const PROXY_URL = "https://ghproxy.com/";

const agentGistRaw = (proxy: string) => {
  const links = document.querySelectorAll<HTMLLinkElement>(
    ".file-actions a, .ml-2:nth-last-child(1) a"
  );

  links.forEach((link) => {
    link.href = proxy + link.href;
  });
};

agentGistRaw(PROXY_URL);

document.addEventListener("pjax:success", () => {
  agentGistRaw(PROXY_URL);
});

export {};
