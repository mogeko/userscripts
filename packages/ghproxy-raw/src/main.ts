const PROXY_URL = "https://ghproxy.com/";

const agentRaw = (proxy: string) => {
  const rawButton = document.querySelector<HTMLLinkElement>("#raw-url");

  if (rawButton) rawButton.href = proxy + window.location.href;
};

agentRaw(PROXY_URL);

document.addEventListener("pjax:success", () => {
  agentRaw(PROXY_URL);
});

export {};
