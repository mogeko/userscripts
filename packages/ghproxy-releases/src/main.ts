const PROXY_URL = "https://ghproxy.com/";

function agentReleases(proxy: string) {
  const svgs = document.querySelectorAll<SVGElement>(
    ".octicon-package, .octicon-file-zip",
  );

  svgs.forEach((svg) => {
    const link = svg.parentNode as HTMLLinkElement;

    link.href = proxy + link.href;
  });
}

agentReleases(PROXY_URL);

document.addEventListener("pjax:success", () => {
  agentReleases(PROXY_URL);
});

export {};
