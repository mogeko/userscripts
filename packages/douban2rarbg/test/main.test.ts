import { describe, it, expect } from "vitest";
import { JSDOM } from "jsdom";

describe("douban2rarbg", () => {
  it("should work", async () => {
    window.document = new JSDOM(
      '<div id="info"><span class="pl">IMDb:</span> tt5057130<br /></div>',
      {
        url: "https://movie.douban.com/subject/26629153",
        referrer: "https://movie.douban.com",
        contentType: "text/html",
      },
    ).window.document;

    await import("../src/main");

    const metaRoot = window.document.querySelector("#info");

    expect(metaRoot).not.toBeNull();
    expect(metaRoot?.textContent).toContain("字幕");
    expect(metaRoot?.textContent).toContain("RARBG");
    expect(metaRoot).toMatchSnapshot();
  });

  it("should pass", () => {
    window.document = new JSDOM('<div id="info"></div>', {
      url: "https://movie.douban.com/subject/26629153",
      referrer: "https://movie.douban.com",
      contentType: "text/html",
    }).window.document;

    import("../src/main");

    const metaRoot = window.document.querySelector("#info");

    expect(metaRoot).not.toBeNull();
    expect(metaRoot?.textContent).not.toContain("字幕");
    expect(metaRoot?.textContent).not.toContain("RARBG");
    expect(metaRoot).toMatchSnapshot();
  });
});
