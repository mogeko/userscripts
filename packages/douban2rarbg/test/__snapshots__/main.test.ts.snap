// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`douban2rarbg > should pass 1`] = `
<div
  id="info"
/>
`;

exports[`douban2rarbg > should work 1`] = `
<div
  id="info"
>
  <span
    class="pl"
  >
    IMDb:
  </span>
   tt5057130
  <br />
  <span>
    <span
      class="pl"
    >
      资源: 
    </span>
    <span
      class="attrs"
    >
      <a
        href="https://rarbg.to/torrents.php?imdb=tt5057130&order=seeders&by=DESC"
        target="_blank"
      >
        RARBG
      </a>
       / 
      <a
        href="https://psa.wf/?s=tt5057130"
        target="_blank"
      >
        PSA
      </a>
       / 
      <a
        href="https://yts.mx/browse-movies/tt5057130/all/all/0/latest/0/all"
        target="_blank"
      >
        Yify
      </a>
       / 
      <a
        href="https://torrentgalaxy.to/torrents.php?search=tt5057130"
        target="_blank"
      >
        TorrentGalaxy
      </a>
       / 
      <a
        href="https://thepiratebay.org/search.php?q=tt5057130"
        target="_blank"
      >
        TPB
      </a>
    </span>
  </span>
  <br />
  <span>
    <span
      class="pl"
    >
      字幕: 
    </span>
    <span
      class="attrs"
    >
      <a
        href="https://www.opensubtitles.org/zh/search/imdbid-5057130/sublanguageid-all/moviename-tt5057130"
        target="_blank"
      >
        opensubtitles
      </a>
       / 
      <a
        href="https://subhd.tv/d/26629153"
        target="_blank"
      >
        SubHD
      </a>
       / 
      <a
        href="https://so.zimuku.org/search?q=tt5057130"
        target="_blank"
      >
        字幕库
      </a>
       / 
      <a
        href="https://r3sub.com/search.php?s=tt5057130"
        target="_blank"
      >
        R3SUB
      </a>
       / 
      <a
        href="http://www.ddzimu.com/download/xslist.php?key=26629153"
        target="_blank"
      >
        点点字幕
      </a>
    </span>
  </span>
  <br />
</div>
`;
